commit f20c51c3e03d8823603e9715e9b0d7d8d65e63ca
Author: Hans Wennborg <hans@chromium.org>
Date:   Tue Apr 29 07:21:24 2025 -0700

    Revert "Enable CREL for arm32 targets"
    
    This reverts commit ffb2d73b50613b01bd73e073d17ab5583db27052.
    
    Reason for revert:
    Users reported arm32 crashes (see bug).
    
    Bug: 376278218
    Original change's description:
    > Enable CREL for arm32 targets
    >
    > CREL was disabled on arm32 because of a reported crash on Linux arm32.
    > However, this platform is not supported and there is currently no
    > evidence of issues on supported arm32 platforms. With no stack trace or
    > crash dump provided it is hard to diagnose the issue.
    >
    > Reenabling CREL allows us to determine if the issue is present on other
    > arm platforms. If so, it will be easier to reproduce the issue. If not,
    > we may hold off on fixing it until more information is provided.
    >
    > Bug: 376278218
    > Change-Id: I3a62f02f564f07be63173b0773b4ecaffbe939b9
    > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6352639
    > Auto-Submit: Collin Baker <collinbaker@chromium.org>
    > Reviewed-by: Arthur Eubanks <aeubanks@google.com>
    > Commit-Queue: Arthur Eubanks <aeubanks@google.com>
    > Cr-Commit-Position: refs/heads/main@{#1434283}
    
    Bug: 376278218
    Change-Id: Iff360b4a1585b19f80c0bd2d396f1ceb19a65d17
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6495023
    Auto-Submit: Hans Wennborg <hans@chromium.org>
    Reviewed-by: Nico Weber <thakis@chromium.org>
    Commit-Queue: Nico Weber <thakis@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#1453213}

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -619,7 +619,7 @@ config("compiler") {
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF
     # (excluding toolchains that use an older version of LLVM).
-    if (is_linux && use_lld && !llvm_android_mainline &&
+    if (is_linux && use_lld && !llvm_android_mainline && current_cpu != "arm" &&
         default_toolchain != "//build/toolchain/cros:target") {
       cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
     }
